standard/expressions.md

```diff
@@ -1464,6 +1510,29 @@ The run-time processing of a delegate invocation of the form `D(A)`, where `D`
 
 See [§20.6](delegates.md#206-delegate-invocation) for details of multiple invocation lists without parameters.
 
+### §null-conditional-invocation-expression Null Conditional Invocation Expression
+
+A *null_conditional_invocation_expression* is syntactically either a *null_conditional_member_access* (§null-conditional-member-access) or *null_conditional_element_access* (§null-conditional-element-access) where the final *captured_access* is an invocation expression ([§12.7.6](expressions.md#1276-invocation-expressions)).
+
+Unlike the syntactically equivalent *null_conditional_member_access* or *null_conditional_element_access*, a *null_conditional_invocation_expression* may be classified as nothing and only occurs as part of a *statement_expression* ([§13.7](statements.md#137-expression-statements)) or *anonymous_function_body* ([§12.16.1](expressions.md#12161-general)).
```

Comment:
You are right, expression-bodied method has been missed.

As currently written the grammar doesn't allow a *null_conditional_invocation_expression* within an expression-bodied method. (However both #7 and this one syntactically accept their respective versions of null conditional access which end in an invocation; the restriction here in #251 that such an access cannot return void is a semantic one.)

I haven't looked at whether this can be accommodated easily in #251's current approach of splitting void handling into different rules vs. the nested bullet semantics (as in #7) – the latter are just messy and this part of #251 was an attempt to make the description easier to follow, but maybe this will turn out to show that cannot be done (at least using seperate rules).

An obvious question occurs, should expression-bodied methods be defined in terms of *expression* or *statement_expression* – if the latter then the issue goes away. (I haven't considered the answer yet.)

Null conditionals are certainly proving “fun” (as in both #7 & #251 have problems 🙁)

Reply:
@gafter – You wrote “Alternatively, perhaps it is overspecified as that is a *statement_expression*…” For a moment that seemed to provide hope, but then I considered it, doesn't an expression bodied member have an *expression* and not a *statement_expression*? In this context the latter might be nice, but I think it is/should be the former.

Looking at this I realised that expression bodied members may not be allowed for elsewhere, will add an Issue.

Reply:
@gafter – your wording suggestion above was resolved differently by truncating the sentence after "as nothing"